pkg/tsdb/loki/loki.go

```diff
@@ -210,8 +211,7 @@ func (s *Service) parseQuery(dsInfo *datasourceInfo, queryContext *backend.Query
 			return nil, err
 		}
 
-		resolution := queryModel.Model.Get("resolution").MustInt64(1)
-		interval := e.intervalCalculator.Calculate(*queryContext.TimeRange, dsInterval)
+		resolution := model.Resolution
```

Comment:
@idafurjes do you know if there's a way to ensure or check that resolution won't be undefined and will always have a number value?

Reply:
so default value of model.Resolution should be 0, if there is no value from the JSON, does that answer your question?

Reply:
@olbo98 @idafurjes you can just check if the `model.Resolution` is 0 (unset).

Reply:
0 could both mean that the value is 0 or the value is missing, there are other possible solutions, maybe the easiest to use a pointer

Reply:
So @olbo98 that would be something like:
```
		resolution := model.Resolution
		
        if resolution == 0 {
			resolution = 1
		}
```

Reply:
see my comment above

Reply:
if resolution is never 0 - in that case the check would work

Reply:
Well in theory the resolution value shouldn't be missing and the default should be 1, at least in front-end, but i don't know how the back-end is connected so i was thinking we could need a check to see if the value may be missing.

Reply:
I understand, what are the possible values for the resolution?

Reply:
the possible values are 1-5 and 10

Reply:
Then checking for 0 is fine :) or even checking if the value is between 1-5 or 10, let me know if you have other questions